### PR TITLE
Improve rendering of lines in 3D (against disappearing in certain angles) issue #691

### PR DESCRIPTION
**Pull request by archmoj**
Previously the angles computed in the `gl-plot3d` module necessary to match rectangle shapes towards camera point were not accurate. Therefore 2D representations of 3D lines were not oriented properly in various perspective angles. Disappearing and tininess impacts resulted from the bug could be clearly observed namely when the thicknesses of lines using `scatter3d` were increased. Please refer to the [issue](https://github.com/plotly/plotly.js/issues/691) #691 for more information.
This PR improves rendering of lines in 3D scenes by performing proper 3D/2D calculations within the webgl vertex shader of the module.
@alexcjohnson

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "gl-contour2d": "^1.1.4",
     "gl-error3d": "^1.0.7",
     "gl-heatmap2d": "^1.0.4",
-    "gl-line3d": "^1.1.2",
+    "gl-line3d": "git://github.com/gl-vis/gl-line3d.git#dfbd94f4290d30097c9b7c9f36d2a241732124c0",
```
Comment (archmoj) [inline package.json]: https://github.com/gl-vis/gl-line3d/compare/master...issue-691

**Comment by alexcjohnson** [comment]
Thanks @archmoj - In the image tests I see a bunch of minor changes but nothing that I can clearly point to and say "yes, that was a bug and now it's fixed." Can you include a new image, like the one posted in #691, to make it obvious what was fixed? Then, since it doesn't generate a diff, post a screenshot in a comment "here's what this image looks like on the master branch, notice the missing line between the top two points" or whatever.

I do think I see something a little odd, the change in [gl3d_set-ranges](https://github.com/plotly/plotly.js/pull/3128/files?short_path=17cbbc9#diff-17cbbc97019f5b261179c860f0dcef28) - the clipped line got a little bit longer, and if I rotate the scene it looks like it's popping out the top just a tiny bit:
screen shot 2018-10-22 at 9 16 39 am
Was that intentional / expected?

**Comment by archmoj** [comment]
Before:
newplot 52
After:
newplot 53

**Comment by alexcjohnson** [comment]
Thanks - that before/after comparison is exactly what I wanted to see. The segment joins definitely look better now 🌟 
#691 also shows an issue of some segments disappearing entirely, or becoming very thin compared to others. Can we include a case like this in the test image too?

**Comment by archmoj** [comment]
Yes I've already added this to the image test. But I would like to possibly add few other traces to better show the changes.

**Comment by archmoj** [comment]
To compare previous and new images:
https://github.com/plotly/plotly.js/pull/3128/commits/8641bed97446d8e7267d7431f2756762ee783c86?short_path=59dc7bd#diff-59dc7bd2fd8bab2b682e77565af6615b

**Comment by archmoj** [comment]
To compare previous gl-line3d with new gl-line3d performance, please view old & new images here:
https://github.com/plotly/plotly.js/pull/3128/commits/420fe722d966193f2416f498472f98100341e3c2?short_path=59dc7bd#diff-59dc7bd2fd8bab2b682e77565af6615b

File: package.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "gl-contour2d": "^1.1.4",
     "gl-error3d": "^1.0.7",
     "gl-heatmap2d": "^1.0.4",
-    "gl-line3d": "^1.1.2",
+    "gl-line3d": "git://github.com/gl-vis/gl-line3d.git#33b9e850de754cde38d859328ae562125edc90f2",
```
Comment (archmoj) [inline package.json]: https://github.com/gl-vis/gl-line3d/pull/7

**Comment by etpinard** [comment]
Closing in favour of https://github.com/plotly/plotly.js/pull/3163

**Pull request closed without merging**